### PR TITLE
Add a loader for nastran '.nas' files.

### PR DESCRIPTION
**Pull request by 16uhrpasing**
Hi,

thank you very much. VSG is an awesome piece of software.
I wanted to give something back by adding a loader for nastran files.

An example project with an example nastran file can be found here: https://github.com/16uhrpasing/VSGNastranLoader
It just needs VSG and vsgXchange as a dependecy. Let me know if you have any questions. During the next weeks I'd also add a README.

Have a great day!

### CODE REVIEW

File: src/nastran/nastran.cpp [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+{
+    auto ext = vsg::lowerCaseFileExtension(filename);
+    if (ext != "nas" || !vsg::fileExists(filename)) return { };
+    
+
+    std::ifstream stream(filename.c_str(), std::ios::in);
+    if (!stream) return { };
+    return read(stream, options);
+
+    return vsg::ref_ptr<vsg::Object>();
```
Comment (robertosfield) [inline src/nastran/nastran.cpp:217]: I presume this line of code is hang up of dev process/debugging.  This will need to be removed.

**Comment by robertosfield** [comment]
Thanks for the submission.  I have done a first pass code view and will get to my dev system later this weekend to test it out. 

I haven't worked with nastram before, is there a spec for the file format?

With the temperature value being passed as a w value, then mapped to as a greyscale colour, it strikes me that this could be utilize a 1D texture for a colour map.  The fragment shader reads the temperature value, maps to a range of the colour map and then uses this as the fragment colour.  

I haven't written a specific colour map class for the VSG yet, but may write one soon, this would be essentially be a map<float, vec4> that is used to fill in a vsg::vec4Array which is assigned to a vsg::DescriptorImage to be used a texture.

The nastram loaded needn't wait for me to complete the colour map support, we could just merge and then further develop it.

**Comment by robertosfield** [comment]
I have done a review and feel the code wasn't ready to merge with master.  I have created a nastranLoader branch and merged the PR with this branch:

[https://github.com/vsg-dev/vsgXchange/tree/nastranLoader](https://github.com/vsg-dev/vsgXchange/tree/nastranLoader)

I then started restructuring the code and rewriting elements of it to make the code more maintainable, fix warnings, and improve the flexibility of using with a standard viewer.  This loader now can be used by using by standard apps like vsgviewer:

   vsgviewer truck.nas

@16uhrpasing could you use this as a base for any further work you do. Thanks.

**Comment by 16uhrpasing** [comment]
Thank you very much for improving the code, I'm relatively new to CG-programming itself as well as OSG and VSG, this gives me further insight on how to structure my code. Unfortunately I'm busy with finishing university for the next weeks, but as soon as I'm finished I will continue to work on the loader. Thank you.

**Pull request closed without merging**